src/Networking/E_Switch.cpp

```diff
@@ -60,12 +60,17 @@ void Switch::packetArrived(Port* inPort, Packet* packet)
 				Packet* newPacket = this->clonePacket(packet);
 				if(drop)
 				{
-					uint16_t false_checksum = 0xEEEE;
-					uint32_t false_data = 0xEEEEEEEE;
-					if(newPacket->getSize() >= (14+20+20))
-						newPacket->writeData(14+20+16, &false_checksum, sizeof(false_checksum));
-					if(newPacket->getSize() >= (14+20+20+4))
-						newPacket->writeData(14+20+20, &false_data, sizeof(false_data));
```

Comment:
If we only corrupt checksum field, an implementation w/o checksum validation will pass the test.

Reply:
Now corrupts data when data exists, otherwise checksum only.